lib/bolt/cli.rb

```diff
@@ -389,7 +389,7 @@ def pal
       @pal ||= Bolt::PAL.new(config.modulepath, config.hiera_config, config.compile_concurrency)
     end
 
-    def validate_file(type, path)
+    def validate_file(type, path, allow_dir = false)
```

Comment:
If we're using allow_dir do we need to recurse through the directory to make sure it's descendants are readable?

Reply:
That makes sense to me.